src/studio/src/designer/backend/Controllers/TextsController.cs

```diff
@@ -0,0 +1,65 @@
+using System;
+using System.Collections.Generic;
+using System.Text.Json;
+using System.Threading.Tasks;
+
+using Altinn.Studio.Designer.Helpers;
+using Altinn.Studio.Designer.Services.Interfaces;
+
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Altinn.Studio.Designer.Controllers
+{
+    /// <summary>
+    /// Controller containing actions related to text files in the
+    /// new format; text.*.json with key:value pairs.
+    /// </summary>
+    /// <remarks>
+    /// NB: Must not be confused with TextController (singular)
+    /// which interacts with the text files in the old format.
+    /// </remarks>
+    [Authorize]
+    [AutoValidateAntiforgeryToken]
+    [Route("designer/api/v2/{org}/{repo}/texts")]
+    public class TextsController : ControllerBase
+    {
+        private readonly ITextsService _textsService;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TextsController"/> class.
+        /// </summary>
+        /// <param name="textsService">The languages service.</param>
+        public TextsController(ITextsService textsService)
+        {
+            _textsService = textsService;
+        }
+
+        /// <summary>
+        /// Endpoint for getting the complete text file for a specific language.
+        /// </summary>
+        /// <param name="org">Unique identifier of the organisation responsible for the app.</param>
+        /// <param name="repo">Application identifier which is unique within an organisation.</param>
+        /// <param name="languageCode">Language identifier specifying the text file to read.</param>
+        /// <returns>Text file</returns>
+        /// <remarks>If duplicates of keys are tried added the
+        /// deserialization to dictionary will overwrite the first
+        /// key:value pair with the last key:value pair</remarks>
+        [HttpGet]
+        [Produces("application/json")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [Route("{languageCode}")]
+        public async Task<ActionResult<Dictionary<string, string>>> GetText(string org, string repo, [FromRoute] string languageCode)
+        {
+            string developer = AuthenticationHelper.GetDeveloperUserName(HttpContext);
+
+            string textContent = await _textsService.GetTextContent(org, repo, developer, languageCode);
+
+            Dictionary<string, string> jsonTextContent =
```

Comment:
Let the service class return Dictionary<string, string> (the model in this case) instead. The less the controller does the better, and others using the service would work on strongly typed models instead of knowing what it should be serialized to.